src/table_factory.h

```diff
@@ -45,15 +49,24 @@ class TitanTableFactory : public TableFactory {
 
   void* GetOptions() override { return base_factory_->GetOptions(); }
 
+  // REQUIRE: mutex held
```

Comment:
Sorry. After a second though I think adding a mutex to guard `mutable_cf_options_` is a better idea. rocksdb flush/compaction job can run in concurrent and call `NewTableBuilder`. Both `NewTableBuilder` and `SetBlobRunMode` need to take the mutex.

Reply:
Isn't that minor compared to frequent use of mutex when accessing different container? Could we optimize that as well, e.g. use multiple spinlocks / make segmented map.

Reply:
what do you mean by "accessing different container"? Adding a mutex is not for performance, it is to make access of `blob_run_mode` thread-safe. RocksDB flush/compaction has no way to take TitanDBImpl::mutex_ before calling `NewTableBuilder`.

Reply:
got it.